openc3-init/plugins/packages/openc3-tool-common/src/components/FileOpenSaveDialog.vue

```diff
@@ -186,29 +197,44 @@ export default {
     },
   },
   created() {
-    Api.get(this.apiUrl)
-      .then((response) => {
-        this.items = []
-        this.id = 1
-        for (let file of response.data) {
-          this.filepath = file
-          this.insertFile(this.items, 1, file)
-          this.id++
-        }
-        if (this.inputFilename) {
-          this.selectedFile = this.inputFilename
-        }
-      })
-      .catch((error) => {
-        this.$emit('error', `Failed to connect to OpenC3. ${error}`)
-      })
+    // Make TEMP_PREFIX available to the template
+    this.TEMP_PREFIX = TEMP_PREFIX
+    this.loadFiles()
     if (this.requireTargetParentDir) {
       Api.get('/openc3-api/targets').then((response) => {
         this.targets = response.data
       })
     }
   },
   methods: {
+    loadFiles: function () {
+      Api.get(this.apiUrl)
+        .then((response) => {
+          this.items = []
+          this.id = 1
+          for (let file of response.data) {
+            this.filepath = file
+            this.insertFile(this.items, 1, file)
+            this.id++
+          }
+          if (this.inputFilename) {
+            this.selectedFile = this.inputFilename
+          }
+          if (this.apiUrl === '/script-api/scripts') {
```

Comment:
Shouldn't script-api/scripts just return these?

Reply:
ie. no extra call to /temp_files needed

Reply:
I thought about that but didn't know if there would be side effects. I can try integrating it into scripts.